File: 658-find-k-closest-elements/find-k-closest-elements.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* findClosestElements(int* arr, int arrSize, int k, int x, int* returnSize) {
    int left=0,right=arrSize-k;
    while(left<right){
        int mid=left+(right-left)/2;
        if(x-arr[mid]>arr[mid+k]-x){
            left=mid+1;
        }
        else{
            right=mid;
        }
    }
    int* result = (int*)malloc(sizeof(int) * k);
    for (int i = 0; i < k; i++) {
        result[i] = arr[left + i];
    }

    *returnSize = k;
    return result;
    
}

File: 658-find-k-closest-elements/find-k-closest-elements.py
class Solution(object):
    def findClosestElements(self, arr, k, x):
        n=len(arr)
        left,right=0,n-k
        while left<right:
            mid=(left+right)//2
            if x-arr[mid]>arr[mid+k]-x:
                left=mid+1
            else:
                right=mid
        return arr[left:left+k]            
        